lib/core.js

```diff
@@ -121,6 +121,10 @@ module.exports = class Core {
         }
       }
 
+      if (header.keyPair && !header.keyPair.secretKey) {
+        header.keyPair.secretKey = null
+      }
```

Comment:
just do it up above when the header is constructed, so we dont mutate a users config object,

```
keyPair: keyPair ? { publicKey: keyPair.publicKey, secretKey: keyPair.secretKey || null} : null
```

Reply:
Good point, fixed